### ISSUE TITLE
Implement aiida-cp2k input scaffolding

### ISSUE THREAD

**Issue by sylviancadars** [cp2k/cp2k-input-tools#29]
Dear cp2k-input-tools developper(s),

I have been trying to use your library to create a python dict from an existing CP2K input, to then use this dict to run new calculations in the aiida-cp2k framework. However I face the following issue.

When I use the CP2KInputParserSimplified parser, the KIND section is in the form : 

'KIND': {'Te': {'BASIS_SET': 'TZVP-MOLOPT-SR-GTH', 'POTENTIAL': 'GTH-BLYP-q6'}, 'O': {'BASIS_SET': 'TZVP-GTH-q6', 'POTENTIAL': 'GTH-BLYP-q6'}}

My problem is that this tree-structure is not understood by AiiDA, which requires a format that seems to correspond to what you call "canonical form", generated by the CP2KInputParser :

'+KIND': [{'BASIS_SET': ['TZVP-MOLOPT-SR-GTH'], 'POTENTIAL': 'GTH-BLYP-q6', '_': 'Te'}, {'BASIS_SET': ['TZVP-GTH-q6'], 'POTENTIAL': 'GTH-BLYP-q6', '_': 'O'}]

except that in this case the '+' signs that are added to most dict key names for a reason that I do not quite understand are problematic. I do not yet know whether aiida-cp2k is able to recognize the key names despite the presence of these + signs, but it will create section duplicates if I try to modify the inoput parameters afterwards within aiida-cp2k.

In summary, is there a way to obtain the tree structure of the 'canonical form' (in particular for the 'KIND' section) without the '+' signs ?

Thanks for your help.
Best regards.

Sylvian

**Comment by dev-zero** [comment]
Yes, compatibility with the aiida-cp2k format is part of the cp2k-input-tools, in fact, there is a already an extension planned in this PR here: https://github.com/aiidateam/aiida-cp2k/pull/105 which allows to create AiiDA `StructureData` from CP2K input files and simply run CP2K input files with AiiDA where the input file is first parsed into an aiida-cp2k parameter dictionary.

The way to do it is this the following:

```
    from cp2k_input_tools.parser import CP2KInputParserSimplified

    parser = CP2KInputParserSimplified(key_trafo=str.upper,
                                       multi_value_unpack=False,  # avoids that something like "BASIS_SET ORB DZVP-MOLOPT-GTH" is unpacked into {"BASIS_SET": ("ORB", "DZVP-MOLOPT-GTH")}
                                       repeated_section_unpack=False,  # prevents the unpacking of repeated sections (removing the list)
                                       level_reduction_blacklist=["KIND"])  # prevents that "KIND H" is turned into {"KIND": {"H": {"BASIS_SET": ...}}} but kept as {"KIND": {"_": "H", "BASIS_SET": ...}}} and with the option above makes it compatible with aiida-cp2k's way of CP2K input representation
    tree = parser.parse(cp2k_input_file)
```

**Comment by dev-zero** [comment]
I've prepared a first PR for a new command `cp2k2aiida` which does (here run via poetry):

```console
$ poetry run cp2k2aiida tests/inputs/NaCl.inp 

from aiida.engine import run
from aiida.orm import (load_code, Dict, StructureData)

cp2k_code = load_code('...')

# Structure
structure = StructureData(...)

# Parameters
parameters = Dict(
    dict={
           "GLOBAL": {
               "PROJECT_NAME": "NaCl",
               "RUN_TYPE": "ENERGY_FORCE",
               "PRINT_LEVEL": "MEDIUM",
           },
           "FORCE_EVAL": {
               "DFT": {
                   "QS": {
                       "EPS_DEFAULT": 1e-12,
                       "METHOD": "GAPW",
                   },
                   "POISSON": {
                       "PERIODIC": "XYZ",
                   },
                   "SCF": {
                       "DIAGONALIZATION": {
                       },
                       "EPS_SCF": 1e-07,
                       "MAX_SCF": 80,
                       "SCF_GUESS": "ATOMIC",
                   },
                   "KPOINTS": {
                       "SCHEME": "MONKHORST-PACK 8 8 8",
                       "FULL_GRID": False,
                       "SYMMETRY": False,
                   },
                   "XC": {
                       "XC_FUNCTIONAL": {
                           "_": "PBE",
                       },
                   },
                   "MGRID": {
                       "REL_CUTOFF": 80.0,
                       "CUTOFF": 800.0,
                       "NGRIDS": 6,
                   },
                   "BASIS_SET_FILE_NAME": "BASIS_pob  ! this is shipped with CP2K 7.0+",
                   "POTENTIAL_FILE_NAME": "ALL_POTENTIALS",
               },
               "SUBSYS": {
                   "CELL": {
                       "A": "5.64123539364476 0.0 0.0",
                       "B": "0.0 5.64123539364476 0.0",
                       "C": "0.0 0.0 5.64123539364476",
                       "PERIODIC": "XYZ",
                   },
                   "COORD": {
                       "SCALED": True,
                       "*": [
                           "Na  0.000000    0.500000    0.500000",
                           "Na  0.000000    0.500000    0.500000",
                           "Na  0.500000    0.000000    0.500000",
                           "Na  0.500000    0.500000    0.000000",
                           "Cl  0.500000    0.500000    0.500000",
                           "Cl  0.500000    0.000000    0.000000",
                           "Cl  0.000000    0.500000    0.000000",
                           "Cl  0.000000    0.000000    0.500000",
                       ],
                   },
                   "KIND": [
                       {
                       "_": "Na",
                       "BS": {
                           "ALPHA": {
                               "N": 3,
                               "L": 0,
                               "NEL": -2,
                           },
                           "BETA": {
                               "N": 3,
                               "L": 0,
                               "NEL": -2,
                           },
                       },
                       "ELEMENT": "Na",
                       "BASIS_SET": "pob-TZVP",
                       "POTENTIAL": "ALL",
                       },
                       {
                       "_": "Cl",
                       "BS": {
                           "ALPHA": {
                               "N": 3,
                               "L": 1,
                               "NEL": 2,
                           },
                           "BETA": {
                               "N": 3,
                               "L": 1,
                               "NEL": 2,
                           },
                       },
                       "ELEMENT": "Cl",
                       "BASIS_SET": "pob-TZVP",
                       "POTENTIAL": "ALL",
                       },
                   ],
               },
               "PRINT": {
                   "STRESS_TENSOR": {
                       "ADD_LAST": "NUMERIC",
                       "FILENAME": "stress_tensor",
                   },
                   "FORCES": {
                       "ADD_LAST": "NUMERIC",
                       "FILENAME": "forces",
                   },
               },
               "METHOD": "QUICKSTEP",
               "STRESS_TENSOR": "ANALYTICAL",
           },
    })

# Construct process builder.
builder = cp2k_code.get_builder()
builder.structure = structure
builder.parameters = parameters
builder.code = cp2k_code
builder.metadata.options.resources = {
    "num_machines": 1,
    "num_mpiprocs_per_machine": 1,
}
builder.metadata.options.max_wallclock_seconds = 1 * 3 * 60

run(builder)

```

**Comment by dev-zero** [comment]
this is now part of the `fromcp2k` as of version v0.6.0:

```
$ fromcp2k --format aiida-cp2k-calc path/to/your/cp2k.inp
```
